docs/sdk/api/apiRequestBuilder.md

```diff
@@ -93,3 +93,78 @@ client.execute(channelsRequest)
 .then(result => ...)
 .catch(error => ...)
 ```
+
+#### Declarative Usage
+
+A declarative API exists as an alternative to the imperative API (shown in the example above).
+
+```js
+// same imports and instantiation as above
+const channelsUri = requestBuilder.channels
+  .parse({
+    where: ['key = "foo"'],
+    perPage: 1,
+    version: 3,
+  })
+  .build()
+```
+
+**Type of Params**
+This declarative `parse` API accepts an object of the following shape:
+
+```js
+{
+  // query-expand
+  expand?: Array<string>;
+
+  // query-id
+  id?: ?string;
+  key?: ?string;
+  customerId?: ?string;
+
+  // query-page
+  sort: Array<{ by: string, direction: 'asc' | 'desc' }>;
+  page: ?number;
+  perPage: ?number;
+
+  // query-projection
+  staged?: boolean;
+  priceCurrency?: string;
+  priceCountry?: string;
+  priceCustomerGroup?: string;
+  priceChannel?: string;
+
+  // query-search
+  text?: ?{
+    language?: string;
+    value?: string;
+  };
+  fuzzy?: boolean;
+  fuzzyLevel?: number;
+  markMatchingVariants?: boolean;
+  facet?: Array<string>;
+  filter?: Array<string>;
+  filterByQuery?: Array<string>;
+  filterByFacets?: Array<string>;
+
+  // query-suggest
+  searchKeywords?: Array<{language: string; value: string;}>;
+
+  // query
+  where?: Array<string>;
+  whereOperator?: 'and' | 'or';
+
+  // version
+  version?: string;
+}
+```
```

Comment:
I fear that this might get outdated fast. Should we rather link to the type definition?

Reply:
I think that makes sense, because the reference to these keys are much more clear and that we have a SSOT.
If there is any deviation to what the original documentation, as in if there is a GET query we don't support on the requestBuilder, we should state the as well

Reply:
@emmenko The `params` are an argument for that one specific `parse` function. They are not used anywhere else. I don't think it should go into the Glossary.

@adnasa I do not understand your comment 😕

Reply:
I think the documentation of the params should be in the documentation as it is. The imperative usage is already there and would make it more succinct

Reply:
what I was saying was __yes__ link to the docs please. :)

Reply:
@adnasa bbbbbbbbut these are the docs? I'm even more confused now 😕

Reply:
Anyways, @wizzy25 and I seem to agree to keep it as-is.

Reply:
I'm talking about this page https://commercetools.github.io/nodejs/sdk/Glossary.html

Reply:
@emmenko I know. But why would we add a glossary definition for the parameters of a single function of a single module to the global `sdk` glossary?

From the description:

> This is a glossary of the core terms around SDK packages, along with their type signatures

I don't consider the `params` argument we accept in `parse` to be a core term. So it should not go in there unless I'm misunderstanding something.

Reply:
I'm fine also keeping it here for now.